### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by 0xStarcat**
- Adds typescript to the build phase
- generates `**/*.d.ts` files for each file and the `dist/index.js` build file and places them into `dist/types` 
- adds `types` field to `package.json`

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 *
 
 # But not these files...
-!dist/index.js
\ No newline at end of file
+!/dist/
+!dist/*
```
Comment (0xStarcat) [inline .npmignore:6]: make sure npm includes the types directory

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (0xStarcat) [inline tsconfig.json:1]: pretty much a boilerplate config

**Comment by 0xStarcat** [comment]
Open to reviewers - specifically if anyone can test if this will allow the library to work in Angular without issues. I don't do Angular

File: package.json [hunk 3]
```diff
@@ -28,13 +28,15 @@
     "moment-timezone": "^0.5.26",
     "tz-lookup": "^6.1.24"
   },
+  "types": "./dist/types/**/**.d.ts",
```
Comment (0xStarcat) [inline package.json:31]: this should link the type files to people's projects :crossed_fingers:

**Comment by 0xStarcat** [comment]
### Open to reviewers - specifically if anyone can test if this will allow the library to work in Angular without issues. I don't do Angular

**Comment by KevinMartins367** [comment]
i can test

**Comment by 0xStarcat** [comment]
> i can test

thank you!

**Comment by KevinMartins367** [comment]
does not work, but add ` "noImplicitAny": false,` in tsconfig.json of the project in question it stops showing the error

**Comment by KevinMartins367** [comment]
therefore, it would not be necessary to change this package

**Comment by KevinMartins367** [comment]
oh my mistake, sorry. Moving index.d.ts to the package's ./dist folder worked perfectly

**Comment by KevinMartins367** [comment]
oh my mistake, sorry. Moving index.d.ts to the package's ./dist folder worked perfectly. but we still have to declare the Origin, Horoscope because I am receiving `The module '" circular-natal-horoscope-js "' has no exported members' Horoscope'.ts (2305)`

**Comment by KevinMartins367** [comment]
I figured out!!! the ./dist/index.d.ts must be the same as ./src/index.d.ts and thus `
import {Origin} from "../src/Origin";
import {Horoscope} from "../src/Horoscope";
export {Origin, Horoscope};

**Comment by 0xStarcat** [comment]
Amazing! I see your pull request #10 and will talk about things there. thank you :)

**Comment by 0xStarcat** [comment]
I managed to test this in an angular project with a local version the package. The fix seemed to be simply to add an `index.d.ts` file containing `declare module 'circular-natal-horoscope-js';` to the `dist` folder and then adding a `typings` key to the package's `package.json` file. 

I added a `tsconfig.json` file and generate a types directory for posterity, but that wasn't the solution to remove the warning in my tests. 

I'm going to merge and do a release shortly to test this out

**Comment by 0xStarcat** [comment]
Had a follow up PR to fix some minor issues https://github.com/0xStarcat/CircularNatalHoroscopeJS/pull/13

**Pull request merged**